From 0dafee3e41cefa1636958880f9c09fcf1da1ca39 Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Thu, 8 Jun 2023 16:13:42 +0800
Subject: [PATCH] MinGW builds: Fix linking on Rust 1.70.0 and later

Like in issue #968, we must link to -lntdll if we are building with Rust 1.70.0
or later.

This should fix the msys CI.
---
 configure.ac | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/configure.ac b/configure.ac
index 23b393c8a..01ac3116e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -195,6 +195,11 @@ AS_IF([test x"$platform" = xWindows], [
   LIBS="$LIBS $USERENV_LIBS"
   AC_CHECK_HEADER([bcrypt.h], [BCRYPT_LIBS=-lbcrypt], [], [#include <windows.h>])
   LIBS="$LIBS $BCRYPT_LIBS"
+  # Link -lntdll if using Rust 1.70.0 or later
+  AX_COMPARE_VERSION([$rust_version],[ge],[1.70.0], [
+    AC_CHECK_HEADER([winternl.h], [WINNTDLL_LIBS=-lntdll], [], [#include <ntdef.h>])
+    LIBS="$LIBS $WINNTDLL_LIBS"
+  ])
 ])
 
 dnl ===========================================================================
-- 
GitLab

